### PR TITLE
Implement simple auth via cookie or header

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Enable auth via `Guardian` and `Argon`.

This PR adds 6 API routes:
- GET /user
- POST /user
- PUT /user
- DELETE /user
- POST /sessions
- DELETE /session

Simple stuff, this follows Guardian's best practices. Routes that need login should use the `logged_in` pipelineon the `router.ex`. Either a user has header or cookie auth, or the request is rejected with 404.

Logout and account deletion mechanisms only remove the token from the Phoenix session (the cookie). JWT tokens are not invalidated still. By default tokens last 4 weeks. On prod they last 1 day. All of this can be adjusted by configuring Guardian.

### CODE REVIEW

File: lib/phoenix_starter/accounts/user.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  @doc false
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, @fields)
+    |> validate_required(@required_fields)
+    |> unique_constraint(:email, downcase: true)
+    |> encrypt_password()
+  end
```
Comment (ZePedroResende) [inline lib/phoenix_starter/accounts/user.ex:24]: Could we add a email regex validation and a minimal password size validation ? Even if they're not perfect its still a little more secure

**Review by ZePedroResende — APPROVED** [review]

File: lib/phoenix_starter/accounts/user.ex (hunk 1 above)
Comment (jfranciscosousa) [reply lib/phoenix_starter/accounts/user.ex]: Did that. Min 6 chars

File: lib/phoenix_starter_web/views/user_view.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+defmodule PhoenixStarterWeb.UserView do
+  use PhoenixStarterWeb, :view
+  alias PhoenixStarterWeb.UserView
+
+  def render("index.json", %{users: users}) do
```
Comment (zediogoviana) [inline lib/phoenix_starter_web/views/user_view.ex:5]: We have the `index` view, and the function to get all users, but we don't have a route or the respective function in  `UserController`. Is it supposed to be "ignored" for now?

**Review by zediogoviana — APPROVED** [review]

File: lib/phoenix_starter_web/views/user_view.ex (hunk 2 above)
Comment (jfranciscosousa) [reply lib/phoenix_starter_web/views/user_view.ex:5]: I left as because it can be useful in the future maybe. I'd just leave it

Comment (jfranciscosousa) [reply lib/phoenix_starter_web/views/user_view.ex:5]: Or maybe remove it. I'll remove it.

**Review by DavideSilva — APPROVED** [review]

**Pull request merged**